plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -160,6 +160,17 @@ public void shutdown()
                 .collect(toImmutableList());
     }
 
+    public void createSchema(String schemaName)
+    {
+        // Put an empty schema collection because MongoDB doesn't support a database without collections
+        client.getDatabase(schemaName).createCollection(schemaCollection);
+    }
+
+    public void dropSchema(String schemaName)
+    {
+        client.getDatabase(schemaName).drop();
```

Comment:
Check if database is empty before drop.

`throw new TrinoException(SCHEMA_NOT_EMPTY, "Schema not empty: " + schemaName);`

Reply:
The logic already exists in engine (`DropSchemaTask`).

Reply:
Thanks.

I see MemoryMetadata performs this check, can you remove that if redundant?